### DISCUSSION TITLE
Start workflow without wating for the end of execution

### DISCUSSION THREAD

**Discussion by hsnsalhi** [elsa-workflows/elsa-core#4411] [Q&A]
hello,
How to start a workflow and immediatly get the instanceId without waiting the execution to finish ?
Thank you

**Comment by sfmskywalker** [comment]
Hi @hsnsalhi ,

In Elsa 3, the way to do it is to generate the workflow instance ID yourself, and then including it as one of the arguments when calling `IWorkflowDispatcher.DispatchAsync`, which requires an argument of type `DispatchWorkflowDefinitionRequest`, which allows you to provide a workflow instance ID that it should use when creating the actual workflow instance.

To generate a unique workflow instance ID, you can use the `IIdentityGenerator` service.

In Elsa 2, when using the `IWorkflowLaunchpad` service, the workflow instance ID is included with the result value, such as `RunWorkflowResult` and `CollectedWorkflow`, depending on which method you invoke.

**Reply by hsnsalhi** [reply]
hello @sfmskywalker and thank you for your reply
I don't understand the parameter activityId needed in StartableWorkflow class.
I start the execution using LaunchPad as follows:

var startableWorkflow = new StartableWorkflow(blueprint, instance, null /*I don't know what to put here*/);
var launchResult = await launchPad.DispatchStartableWorkflowAsync(startableWorkflow, input);

**Reply by sfmskywalker** [reply]
Hi @hsnsalhi 

I think you can just leave that to be null - the workflow engine will find the correct activity to schedule, which is the first activity in its list.

**Reply by hsnsalhi** [reply]
Hi @sfmskywalker 

Can you give me an example of how we can use these classes?

When I try as mentioned above, I get the instance Id immediately, but the instance remains in an idle state and execution is never triggered afterward.

**Reply by sfmskywalker** [reply]
Hi @hsnsalhi 

I'm afraid that I don't have any existing examples, except for how these types are used in the source code itself of course.
But, another approach you could try, even if just to see if it works, is to use the `FindStartableWorkflowsAsync` method to find the workflow you want to execute, and use its result to then call `DispatchStartableWorkflowAsync`.
